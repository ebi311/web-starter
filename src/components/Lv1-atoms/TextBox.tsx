import React from 'react';
import { styled } from '../Theme';

const Input = styled.input`
  border: 1px solid ${p => p.theme.surface};
  border-top-style: none;
  border-left-style: none;
  border-right-style: none;
  padding: 0.25em;
`;

const TextBox: React.FC<React.InputHTMLAttributes<
  HTMLInputElement
>> = props => {
  return <Input {...props} type="text" />;
};

export default TextBox;
