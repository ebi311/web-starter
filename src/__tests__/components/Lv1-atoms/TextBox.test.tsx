import React from 'react';
import TextBox from '../../../components/Lv1-atoms/TextBox';
import { shallow } from 'enzyme';

describe('TextBox', () => {
  test('success', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <TextBox value="text" onChange={onChange} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('change');
    expect(onChange).toBeCalled();
  });
});
