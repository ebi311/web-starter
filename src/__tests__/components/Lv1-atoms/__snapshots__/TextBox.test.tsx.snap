// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox success 1`] = `
<input
  className="sc-AxjAm bpwMoe"
  onChange={[MockFunction]}
  type="text"
  value="text"
/>
`;
